rmgpy/kinetics/diffusionLimited.py

```diff
@@ -39,23 +42,32 @@ def getEffectiveRate(self, reaction, T):
         Keq = reaction.getEquilibriumConstant(T) # Kc
         k_reverse = k_forward / Keq
         k_eff = k_forward
-        
         if reactants == 1:
             if products == 1:
                 k_eff = k_forward
             else: # two products; reverse rate is limited
-                k_diff = self.getDiffusionLimit(T, reaction, forward=False)
-                k_eff_reverse = k_reverse*k_diff/(k_reverse+k_diff)
-                k_eff = k_eff_reverse * Keq
+                if not any([prod.isSolvent for prod in reaction.products]): # none of the products is the solvent species. Thus, the reverse rate is limited
+                    k_diff = self.getDiffusionLimit(T, reaction, forward=False)
+                    k_eff_reverse = k_reverse*k_diff/(k_reverse+k_diff)
+                    k_eff = k_eff_reverse * Keq
         else: # 2 reactants
             if products == 1 or products == 3:
-                k_diff = self.getDiffusionLimit(T, reaction, forward=True)
-                k_eff = k_forward*k_diff/(k_forward+k_diff)
+                if not any([react.isSolvent for react in reaction.reactants]): # none of the reactants is the solvent species. Thus, the forward rate limited
+                    k_diff = self.getDiffusionLimit(T, reaction, forward=True)
+                    k_eff = k_forward*k_diff/(k_forward+k_diff)
             else: # 2 products
-                if Keq > 1.0: # forward rate is faster and thus limited
+                if not any([prod.isSolvent for prod in reaction.products]) and not any([react.isSolvent for react in reaction.reactants]): # ensuring that the solvent species is not involved in the rxn
```

Comment:
could you perhaps extract variables out of the many `any([XXX.isSolvent for XXX...` expressions and re-use them in the conditionals?